Documentation/gettingstarted/taints.rst

```diff
@@ -0,0 +1,87 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _taint_effects:
+
+#####################################################
+Considerations on node pool taints and unmanaged pods
+#####################################################
+
+Depending on the environment or cloud provider being used, a CNI plugin and/or
+configuration file may be pre-installed in nodes belonging to a given cluster
+where Cilium is being installed or already running. Upon starting on a given
+node, and if it is intended as the exclusive CNI plugin for the cluster, Cilium
+does its best to take ownership of CNI on the node. However, a couple situations
+can prevent this from happening:
+
+* Cilium can only take ownership of CNI on a node after starting. Pods starting
+  before Cilium runs on a given node may get IPs from the pre-configured CNI.
+
+* Some cloud providers may revert changes made to the CNI configuration by
+  Cilium during operations such as node reboots, updates or routine maintenance.
+
+This is notably the case with GKE and some other cloud providers, in which node
+reboots and upgrades will undo changes made by Cilium and re-instate the default
+CNI configuration.
+
+To help overcome this situation to the largest possible extent in environments
+and cloud providers where Cilium isn't supported as the single CNI, Cilium can
+manipulate Kubernetes's `taints <https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/>`_
+on a given node to help preventing pods from starting before Cilium runs on said
+node. The mechanism works as follows:
+
+1. The cluster administrator places a taint with key
+   ``node.cilium.io/agent-not-ready`` on a given uninitialized node. Depending on
+   the taint's effect (see below), this prevents pods that don't have a matching
+   toleration from either being scheduled or altogether running on the node until
+   the taint is removed.
+
+2. Cilium runs on the node, initializes it and, once ready, removes the
+   ``node.cilium.io/agent-not-ready`` taint.
+
+3. From this point on, pods will start being scheduled and running on the node,
+   having their networking managed by Cilium.
+
+The taint's effect should be chosen taking into account the following
+considerations:
+
+* If ``NoSchedule`` is used, pods won't be *scheduled* to a node until Cilium
+  has the chance to remove the taint. However, one practical effect of this is
+  that if some external process (such as a reboot) resets the CNI configuration on
+  said node, pods that were already scheduled will be allowed to start
+  concurrently with Cilium when the node next reboots, and hence may become
+  unmanaged and have their networking being managed by another CNI plugin.
+
+* If ``NoExecute`` is used, pods won't be *executed* on a node until Cilium has
```

Comment:
Let's add something like:

_won't be executed on a node ..._ but they will be scheduled [1].

and then [1] can point to https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/

Reply:
Looks good!